Tcs 69 rdb should log errors with filtering on subscription and shut down

AdamBrown2 opened the pull request:
Added systemtables.csv for tables which will not be filtered like logmsg and packets.
Added ignoredtables.csv for tables which will not be subscribed too.

Updated the subscription function .stpps.subsegment in datastripe.q to check first if the table is in systemtables.csv, then check if the table has valid filter defined in segmenting.csv and filtermap.csv and if the filter is invalid check if it is in the ingoredtables.csv

If the table is ignored the function will return a message in the form of a dictionary saying that this table has not been subscribed.
If the table is not in ignoredtables and does not have filter an error message as dictionary will be logged and the rdb will go down.

Updated checkvalidfilters in subscriptions.q which checks schemas are valid for each table and shuts the rdb down if there is an issue. This doesn't shut down whenever an ignored table gives an invalid schema.

Added to rdb.q when datastriping is turned to update the list .rdb.ignorelist to include tables that are of type 99 so that the reload function can ignore these tables and work normally.

jonathonmcmurray reviewed (commented):
Haven't gone through all of the changes but a few comments here.

It seems to me like there are a number of seemingly unrelated changes together in this PR, is that the case? Or are they all linked in some way? My preference is strongly for smaller PRs that essentially contain "one change" rather than a bunch of changes grouped together

AdamBrown2 commented on code/common/pubsub.q:
```diff
@@ -160,36 +160,22 @@ init:{[t]
-
-.ds.map:{[numseg;sym] sym@/:group .ds.lookup[numseg] .ds.modmd5 sym}
+// use mod to stripe into number of segments
+.ds.map:{[numseg;sym] sym!(sum each string sym)mod numseg};
```
That is a change Maurice made which I think has been merged into TorQ master branch but not into torq5_develop.

jonathonmcmurray replied:
Could that be put into a separate PR maybe? It seems unrelated to the rest of the changes here

AdamBrown2 replied:
Yes i'll try and undo the changes.

AdamBrown2 commented on code/common/subscriptions.q:
```diff
@@ -62,7 +62,7 @@ replay0:{[tabs;realsubs;schemalist;logfilelist;filters]
   // only use tables user has access to
   subtabs:realsubs[`subtabs];
-  if[count where nullschema:0=count each schemalist;
+  if[count where nullschema:0=count each schemalist; // change needed here 
```
That was an issue Rian spotted that i never got round to working on. I think the if statement was either always true or always false.

AdamBrown2 commented on code/processes/wdb.q:
```diff
@@ -203,27 +203,32 @@ endofdaysave:{[dir;pt]
-		.wdb.flushend[];
-		.wdb.d:()!()];
-	};
+	/-insert process reload outcome into .wdb.reloadsummary 
+        .wdb.reloadsummary[.z.w]:x;
```
I did not work on the wdb.q code i think again this stuff to be merged from the master branch. I will revert it for this pr

AdamBrown2 commented on code/segmentedtickerplant/datastripe.q:
```diff
@@ -38,21 +40,30 @@ segmentfilter:{[tbl;segid]
-     .ps.subtablefiltered[string[tbl];filter;""]
+     $[tbl in .stpps.systemtables;[.stpps.suball[tbl]];
+     [if[filter~"";$[tbl in .stpps.ignoredtables;
+                    [.lg.o[`sub;m:"Table ",string[tbl]," is in ignoredtables.csv and will not be subscribed to"];
+                    :(tbl;`err`msg!(`ignoredtable;m));];
```
Yes that's how it works and i realise now that could be confusing ill make that change.

AdamBrown2 commented on docs/utilities.md:
```diff
@@ -721,84 +721,55 @@ The advantages of striping include performance and throughput improvements. Segm
 The reduction in data access in each process cumulatively multiplies the data throughput by the number of processes. It also allows the process to complete its task faster and without interruption, thereby reducing latency.
 
-#### Example of data striping in TorQ
+**Example of data striping in TorQ**
```
I think this somebody else's work that has not been merge but is in this branch.

AdamBrown2 commented:
I made the changes requested and reverted the pubsub.q. Trying find the where the wdb.q changes came from.

AdamBrown2 commented:
systemstables.csv renamed to fullsubscriptions.csv.
Removed the dictionary error for ignored tables so when a table is ignored it is not defined in the rdb.
Updated rdb.q so that when datastriping is turned on .rdb.subtables is only the tables that have subscribed.
